### DISCUSSION TITLE
Missing CMAKE_SYSTEM_PROCESSOR variable in cross-compiling

### DISCUSSION THREAD

**Discussion by shibayan** [microsoft/vcpkg#17832] [Ideas]
`CMAKE_SYSTEM_PROCESSOR` is not set when building an ARM / ARM64 target, so configure does not work correctly with some libraries. (e.g opencv)

For ARM64 Windows 10 (not UWP), `CMAKE_SYSTEM_PROCESSOR` and `CMAKE_CROSSCOMPILING` is not set.

**Proposed solution**

Set `CMAKE_SYSTEM_PROCESSOR` and `CMAKE_CROSSCOMPILING` when building ARM / ARM64 target using toolchain file.

I created a branch to check the operation.

- Explicitly set `CMAKE_SYSTEM_PROCESSOR` only for ARM / ARM64
- For ARM64 Windows 10 (not UWP), set `CMAKE_SYSTEM_NAME` and `CMAKE_SYSTEM_VERSION`
  - (to set `CMAKE_SYSTEM_PROCESSOR` and `CMAKE_CROSSCOMPILING`)
- Turn off `CMAKE_CROSSCOMPILING` for x86 / x64 UWP

https://github.com/shibayan/vcpkg/tree/crosscompile

**Additional context**

Share the OpenCV 4 configure log. In the modified toolchains, the CPU is correctly detected.

command : `vcpkg install opencv4[core]:arm64-uwp`

**Configure log with master branch**

```
-- Detecting C compile features
-- Detecting C compile features - done
-- Detected processor: 
-- Performing Test HAVE_CXX11 (check file: cmake/checks/cxx11.cpp)
-- Performing Test HAVE_CXX11 - Success
```

**Configure log with crosscompile branch**

```
-- Detecting C compile features
-- Detecting C compile features - done
-- Detected processor: ARM64
-- Performing Test HAVE_CXX11 (check file: cmake/checks/cxx11.cpp)
-- Performing Test HAVE_CXX11 - Success
```

Rel #3036 #4907

**Comment by driver1998** [comment]
We should probably turn off CMAKE_SYSTEM_CROSSCOMPILING when running **vcpkg on ARM64** (if vcpkg actually works on a ARM64 machine).

actually:
ARM64/ARM/x86 on ARM64 -> CMAKE_SYSTEM_CROSSCOMPILING off
x64 on ARM64 -> CMAKE_SYSTEM_CROSSCOMPILING on
